Tests/StreamChatUITests/SnapshotTests/ChatMessageList/Poll/PollAttachmentView_Tests.swift

```diff
@@ -0,0 +1,133 @@
+
+final class PollAttachmentView_Tests: XCTestCase {
+    /// Static setUp() is only run once. Which is what we want in this case to preload the images.
+    override class func setUp() {
+        /// Dummy snapshot to preload the TestImages.yoda.url image
+        /// This was the only workaround to make sure the image always appears in the snapshots.
+        let view = UIImageView(frame: .init(center: .zero, size: .init(width: 100, height: 100)))
+        Components.default.imageLoader.loadImage(into: view, from: TestImages.yoda.url)
+        AssertSnapshot(view, variants: [.defaultLight])
+    }
+
+    let currentUser = ChatUser.mock(
+        id: .unique,
+        imageURL: TestImages.yoda.url
+    )
+
+    func test_appearance() {
+        let poll = makePoll(isClosed: false)
+        let view = makeMessageView(for: poll)
+        AssertSnapshot(view)
+    }
+
+    func test_appearance_whenClosed() {
+        let poll = makePoll(isClosed: true)
+        let view = makeMessageView(for: poll)
+        AssertSnapshot(view, variants: [.defaultLight, .defaultDark])
+    }
+
+    func test_subtitleText() {
+        let pollAttachmentView = PollAttachmentView()
+        let pollDefault = makePoll(isClosed: false)
+        let pollClosed = makePoll(isClosed: true)
+        let pollUniqueVotes = makePoll(isClosed: false, enforceUniqueVote: true)
+        let pollMaxVotesAllowed = makePoll(isClosed: false, maxVotesAllowed: 3)
+
+        pollAttachmentView.content = .init(poll: pollDefault, currentUserId: .unique)
+        XCTAssertEqual(pollAttachmentView.subtitleText, "Select one or more")
+
+        pollAttachmentView.content = .init(poll: pollClosed, currentUserId: .unique)
+        XCTAssertEqual(pollAttachmentView.subtitleText, "Vote ended")
+
+        pollAttachmentView.content = .init(poll: pollUniqueVotes, currentUserId: .unique)
+        XCTAssertEqual(pollAttachmentView.subtitleText, "Select one")
+
+        pollAttachmentView.content = .init(poll: pollMaxVotesAllowed, currentUserId: .unique)
+        XCTAssertEqual(pollAttachmentView.subtitleText, "Select up to 3")
+    }
+}
+
+// MARK: - Factory Helpers
+
+extension PollAttachmentView_Tests {
+    private func makePoll(
+        isClosed: Bool,
+        enforceUniqueVote: Bool = false,
+        maxVotesAllowed: Int? = nil,
+        createdBy: ChatUser? = nil
+    ) -> Poll {
+        Poll.mock(
+            enforceUniqueVote: enforceUniqueVote,
+            name: "Best football player",
+            voteCount: 6,
+            voteCountsByOption: [
+                "ronaldo": 3,
+                "eusebio": 2,
+                "pele": 1,
+                "maradona": 0,
+                "messi": 0
+            ],
+            isClosed: isClosed,
+            maxVotesAllowed: maxVotesAllowed,
+            createdBy: createdBy ?? currentUser,
+            options: [
+                .init(id: "messi", text: "Messi"),
+                .init(id: "ronaldo", text: "Ronaldo", latestVotes: [
```

Comment:
We are talking here about the Brazilian Ronaldo R9, right? If that's the case, totally agree 😃